### ISSUE TITLE
Imported classes aren't recognized as types in --checkJs

### ISSUE THREAD

**Issue by Zarel** [Microsoft/TypeScript#15720]
**TypeScript Version:**  2.3.2

**Code**

test-externclassdef.js:
```js
class Foo {
  constructor() {}
}

exports.Foo = Foo;
```

test-externclass.js:
```js
const {Foo} = require('./test-externclassdef');
const Stuff = require('./test-externclassdef');

/** @type {Foo} */
let a = new Foo();

/** @type {Stuff.Foo} */
let b = new Stuff.Foo();
```

**Expected behavior:**
No error

**Actual behavior:**

> test-externclass.js(5,12): error TS2304: Cannot find name 'Foo'.
>
> test-externclass.js(8,12): error TS2503: Cannot find namespace 'Stuff'.

It's possible I'm forgetting something I need to do, but I can't find any documentation on how this is supposed to work.

At the very least, these error messages are very uninformative. TypeScript clearly recognizes `Foo` as a class, because the next line doesn't give an error.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/14056

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
